uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequestCalendarFeedConfig.php

```diff
@@ -232,4 +232,185 @@ public static function getValuesArray($id) {
       return [];
     }
   }
+
+  /**
+   * Implementation of the addSelectWhereClause:
+   * This ensures that the administrator (user with 'can administer calendar feeds')
+   * permission can access all feed configs while users without this permission can
+   * only access feed configs which they have visibility access to. i.e if they belong
+   * to any of the department or location visibility set  for a feed config.
+   */
+  public function addSelectWhereClause() {
+    if (CRM_Core_Permission::check([['can administer calendar feeds']])) {
+      return;
+    }
+
+    $limitedVisibilityFeedConfigs = $this->getLimitedVisibilityFeedConfigs();
+
+    if (empty($limitedVisibilityFeedConfigs)) {
+      return;
+    }
+
+    $accessibleDepartments = $this->getAccessibleDepartmentsForLoggedInContact();
+    $accessibleLocations = $this->getAccessibleLocationsForLoggedInContact();
+
+    $inaccessibleFeedConfigs = $this->getNotAccessibleFeedConfigs(
```

Comment:
since you're using `inaccessible`  here it might make more sense to rename the `getNotAccessibleFeedConfigs` to use `inaccessible` too